src/luarocks/fs/lua.lua

```diff
@@ -550,10 +551,110 @@ end
 end
 
 ---------------------------------------------------------------------
+-- Download function
+---------------------------------------------------------------------
+
+-- Default to native downloader
+local download_handlers = setmetatable({}, {__index = function()
+      return fs.use_downloader
+end;})
+
+--- Download a remote file.
+-- @param url string: URL to be fetched.
+-- @param filename string or nil: this function attempts to detect the
+-- resulting local filename of the remote file as the basename of the URL;
+-- if that is not correct (due to a redirection, for example), the local
+-- filename can be given explicitly as this second argument.
+-- @return (boolean, string): true and the filename on success,
+-- false and the error message on failure.
+function fs_lua.download(url, filename, cache)
+   assert(type(url) == "string")
+   assert(type(filename) == "string" or not filename)
+   filename = fs.absolute_name(filename or dir.base_name(url))
+   local scheme = dir.split_url(url)
+   return download_handlers[scheme](url, filename, cache)
+end
+
+-- Try lua-http first
+if http_request_ok then
```

Comment:
Is lua-http interface going to be as stable as luasocket or maybe a version check should be added here?